go/store/nbs/store.go

```diff
@@ -634,12 +634,12 @@ func (nbs *NomsBlockStore) addChunk(ctx context.Context, ch chunks.Chunk, addrs
 		}
 		nbs.tables = ts
 		nbs.mt = newMemTable(nbs.mtSize)
-		ok = nbs.mt.addChunk(a, ch.Data())
+		addChunkRes = nbs.mt.addChunk(a, ch.Data())
 	}
-	if ok {
+	if addChunkRes == chunkAdded {
 		nbs.mt.addChildRefs(addrs)
 	}
-	return ok, nil
+	return true, nil
```

Comment:
To maintain the previous semantics, I think:

```suggestion
	return addChunkRes != chunkNotAdded, nil
```

or

```suggestion
	return addChunkRes == chunkAdded || addChunkRes == chunkExists, nil
```

For now I think the only time that would be hit is if a chunk exceeded maxMemTableSize, which would be a whole set of different problems in itself...

Reply:
Going with the second approach.